django/applications/catmaid/static/js/widgets/landmark-widget.js

```diff
@@ -2520,10 +2597,17 @@
           // Load selected CSV files and enable import button if this worked
           // without problems.
           let parsePromises = [];
-          for (let i=0; i<widget.filesToImport.length; ++i) {
-            let file = widget.filesToImport[i];
-            let promise = CATMAID.parseCSVFile(file, ',',
-                widget.importCSVLineSkip, hasFourElements);
+          for (let file of widget.filesToImport) {
+            let promise = CATMAID.parseCSVFile(
+              file, ',', widget.importCSVLineSkip
+            ).then(
+              (rows) => rows.reduce((out, row) => {
```

Comment:
Single function to skip short rows and trim long rows, rather than filtering short rows in a `parseCSVFile` argument and then trimming in a second iteration